### PR TITLE
sign certificates endpoint

### PR DESCRIPTION
**Pull request by ssoroka**
## Summary

sign certificates. bare bones endpoint that doesn't have checks around it yet, as those depend on later steps.

## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [ ] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

closes https://github.com/infrahq/infra/issues/596

### CODE REVIEW

**Review by BruceMacD — COMMENTED** [review]
Can we use this for connector to server mTLS? I'd like to see some basic usage.

File: internal/server/handlers.go [hunk 1]
```diff
@@ -642,6 +645,38 @@ func (a *API) UpdateIdentityInfoFromProvider(c *gin.Context) error {
 	return access.UpdateUserInfoFromProvider(c, info, user, provider)
 }
 
+func (a *API) SignCertificate(c *gin.Context, certReq *api.CertificateSigningRequest) (*api.CertificateSigningResponse, error) {
+	// TODO: check to see if this cert is on the list of approved certs. Until then this function signs any cert.
```
[the remark this answers is not in the document]
Comment (ssoroka) [reply internal/server/handlers.go:649]: no worse than the current process of trusting everything? :D

File: api/certificates.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package api
+
+type CertificateSigningRequest struct {
+	CertificatePEM []byte `json:"certificatePEM" note:"The PEM-encoded certificate to be signed, not including the private key." validate:"required"`
```
[the remark this answers is not in the document]
Comment (ssoroka) [reply api/certificates.go:4]: It errors. We don't want them to do that :D

**Review by BruceMacD — APPROVED** [review]

**Comment by ssoroka** [comment]
Closing this for now. Will re-evaluate whether this is relevant in 3 months

**Pull request closed without merging**